### PR TITLE
Added QuickSort

### PR DESCRIPTION
**Pull request by kaitinghh**

### CODE REVIEW

**Review by 4ndrelim — CHANGES_REQUESTED** [review]
Overall great, minor changes.

File: src/algorithms/sorting/quickSort/hoares/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+### Presence of Duplicates
+The above analysis assumes the absence of duplicates in our array. We can change the implementation slightly to keep 
+all elements = pivot to the left of the pivot element. In this case, if there are many duplicates in the array, 
+e.g. {1, 1, 1, 1}, the 1st pivot will be placed in the 3rd idx, and 2nd pivot in 2nd idx, 3rd pivot in the 1st idx and 
+4th pivot in the 0th idx. As we observe, the presence of many duplicates in the array leads to extremely unbalanced 
+partitioning, leading to a O(n^2) time complexity.
+
+### Usual Implementation of Hoare's QuickSort
+The usual implementation of Hoare's partition scheme does not necessarily put the pivot in its correct position. It 
+merely partitions the array into <= pivot and >= pivot portions.
```
[the remark this answers is not in the document]
Comment (kaitinghh) [reply src/algorithms/sorting/quickSort/hoares/README.md:51]: The description and analysis above assumes no duplicates. But the implementation/code itself can handle duplicates (puts them to left of pivot).

File: src/algorithms/sorting/quickSort/hoares/QuickSort.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+     * @param start the starting index (inclusive) of the sub-array to be partitioned.
+     * @param end   the ending index (inclusive) of the sub-array to be partitioned.
+     * @return the index of the pivot element in its correct position after partitioning.
+     */
+    private static int partition(int[] arr, int start, int end) {
+        int pivot = arr[start];
+        int low = start + 1;
+        int high = end;
+
+        while (low <= high) {
```
[the remark this answers is not in the document]
Comment (kaitinghh) [reply src/algorithms/sorting/quickSort/hoares/QuickSort.java]: mm yeah, it was necessary bc of the way I implemented previously. but I changed to another implementation (increment/decrement pointer before comparing element value w pivot), so now low < high is sufficient

File: src/algorithms/sorting/quickSort/hoares/README.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+The pivot is in the correct position, with elements to its left being < it, and elements to its right being > it.
+
+![Lecture Hoare's QuickSort](../../../../../assets/LectureHoares.jpeg)
+Example Credits: Prof Seth/Lecture Slides
+
+## Complexity Analysis
+Complexity Analysis: (this analysis is based on fixed index pivot selection)
+
+Time:
```
[the remark this answers is not in the document]
Comment (kaitinghh) [reply src/algorithms/sorting/quickSort/hoares/README.md:23]: I think is O(logn)?

**Review by 4ndrelim — APPROVED** [review]
LGTM, but pending some comments from kaiting.

**Pull request merged**